java/serve/src/main/java/io/ray/serve/router/ReplicaSet.java

```diff
@@ -54,26 +77,30 @@ public ReplicaSet(String deploymentName) {
 
   @SuppressWarnings("unchecked")
   public synchronized void updateWorkerReplicas(Object actorSet) {
-    List<String> actorNames = ((ActorNameList) actorSet).getNamesList();
-    Set<ActorHandle<RayServeWrappedReplica>> workerReplicas = new HashSet<>();
-    if (!CollectionUtil.isEmpty(actorNames)) {
-      actorNames.forEach(
-          name ->
-              workerReplicas.add(
-                  (ActorHandle<RayServeWrappedReplica>)
-                      Ray.getActor(name, Constants.SERVE_NAMESPACE).get()));
-    }
-
-    Set<ActorHandle<RayServeWrappedReplica>> added =
-        new HashSet<>(Sets.difference(workerReplicas, inFlightQueries.keySet()));
-    Set<ActorHandle<RayServeWrappedReplica>> removed =
-        new HashSet<>(Sets.difference(inFlightQueries.keySet(), workerReplicas));
-
-    added.forEach(actorHandle -> inFlightQueries.put(actorHandle, Sets.newConcurrentHashSet()));
-    removed.forEach(inFlightQueries::remove);
-
-    if (added.size() > 0 || removed.size() > 0) {
-      LOGGER.info("ReplicaSet: +{}, -{} replicas.", added.size(), removed.size());
+    if (null != actorSet) {
+      Set<String> actorNameSet = new HashSet<>(((ActorNameList) actorSet).getNamesList());
+      Set<String> added = new HashSet<>(Sets.difference(actorNameSet, inFlightQueries.keySet()));
+      Set<String> removed = new HashSet<>(Sets.difference(inFlightQueries.keySet(), actorNameSet));
+      added.forEach(
+          name -> {
+            try {
+              Optional<BaseActorHandle> handleOptional =
+                  Ray.getActor(name, Constants.SERVE_NAMESPACE);
+              if (handleOptional.isPresent()) {
+                actorHandlerMap.put(name, handleOptional.get());
+                inFlightQueries.put(name, Sets.newConcurrentHashSet());
+              } else {
+                LOGGER.warn("Can not get actor handle. actor name is {}", name);
+              }
+            } catch (Throwable e) {
```

Comment:
I don't think we should catch this exception because it will not raise any.

Reply:
Delete try-catch code.  But the API comment is wrong.
![image](https://user-images.githubusercontent.com/11265783/182785279-cc747bea-06b8-4d14-a30e-38994c0c67df.png)